packages/node_modules/@ciscospark/plugin-memberships/src/memberships.js

```diff
@@ -24,6 +39,177 @@ import {SparkPlugin, Page} from '@ciscospark/spark-core';
+      .then((person) => {
+        that.eventEnvelope.createdBy = person.id;
+        that.eventEnvelope.orgId = person.orgId;
+      }).catch((e) => {
+        console.error(`Unable to get person info for memberships \
+          event envelope ${e.message}`);
+      });
+
+    // Register to listen to events
+    return this.spark.internal.mercury.connect()
+      .then(() => this.listenTo(this.spark.internal.mercury,
+        WEBEX_EVENT.TEAMS_ACTIVITY,
+        (event) => this.onWebexApiEvent(event)));
+  },
+
+  /**
+   * Trigger a membership related events.
+   * @param {Object} event
+   * @returns {undefined} -- nothing //linter requires return in JSDoc
+   */
+  onWebexApiEvent(event) {
+    const {activity} = event.data;
+
+    /* eslint-disable no-case-declarations */
+    switch (activity.verb) {
+      case API_ACTIVITY_VERB.ADD:
+        const membershipCreatedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.CREATED);
+
+        if (membershipCreatedEventData) {
+          debug(`membership "created" payload: \
+            ${JSON.stringify(membershipCreatedEventData)}`);
+          this.trigger(SDK_EVENT.CREATED, membershipCreatedEventData);
+        }
+        break;
+
+      case API_ACTIVITY_VERB.LEAVE:
+        const membershipDeletedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.DELETED);
+
+        if (membershipDeletedEventData) {
+          debug(`membership "deleted" payload: \
+            ${JSON.stringify(membershipDeletedEventData)}`);
+          this.trigger(SDK_EVENT.DELETED, membershipDeletedEventData);
+        }
+        break;
+
+      case API_ACTIVITY_VERB.ADD_MODERATOR:
+      case API_ACTIVITY_VERB.REMOVE_MODERATOR:
+      case API_ACTIVITY_VERB.HIDE:
+        const membershipUpdatedEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.UPDATED);
+
+        if (membershipUpdatedEventData) {
+          debug(`membership "updated" payload: \
+            ${JSON.stringify(membershipUpdatedEventData)}`);
+          this.trigger(SDK_EVENT.UPDATED, membershipUpdatedEventData);
+        }
+        break;
+
+      case API_ACTIVITY_VERB.ACKNOWLEDGE:
+        const membershipSeenEventData =
+          this.getMembershipEvent(activity, SDK_EVENT.SEEN);
+
+        if (membershipSeenEventData) {
+          debug(`membership "updated" payload: \
+            ${JSON.stringify(membershipSeenEventData)}`);
+          this.trigger(SDK_EVENT.SEEN, membershipSeenEventData);
+        }
+        break;
+
+      default:
+        break;
+    }
+  },
+
+  /**
+   * Constructs the data object for an event on the memberships resource,
+   * adhering to Hydra's Webehook data structure memberships.
+   * @param {Object} activity from mercury
+   * @param {Object} event type of "webhook" event
+   * @returns {Object} constructed event
+   */
+  getMembershipEvent(activity, event) {
+    const sdkEvent = _.cloneDeep(this.eventEnvelope);
+    let member = '';
+    let space = '';
```

Comment:
Is there a case where it's valid for these to be empty strings? If not, declare as `let member;`, etc.

Reply:
Will do